### PR TITLE
fix building wasm repl

### PR DESCRIPTION
**Pull request by Anton-4**
Original error:
```
> wasm-pack build --profiling --target web -- --features console_error_panic_hook
2022-12-23T09:13:18.6004478Z [INFO]: Checking for the Wasm target...
2022-12-23T09:13:18.6429592Z [INFO]: Compiling to Wasm...
2022-12-23T09:13:18.9558246Z    Compiling libc v0.2.135
2022-12-23T09:13:18.9558748Z    Compiling version_check v0.9.4
2022-12-23T09:13:18.9559487Z    Compiling proc-macro2 v1.0.40
2022-12-23T09:13:18.9560051Z    Compiling cfg-if v1.0.0
2022-12-23T09:13:18.9560660Z    Compiling unicode-ident v1.0.1
...
2022-12-23T09:13:34.0773740Z    Compiling matchers v0.1.0
2022-12-23T09:13:34.1658207Z error[E0425]: cannot find value `EISDIR` in crate `libc`
2022-12-23T09:13:34.1658825Z  --> /home/big-ci-user/.cargo/registry/src/github.com-1ecc6299db9ec823/remove_dir_all-0.7.0/src/unix.rs:8:50
2022-12-23T09:13:34.1658913Z   |
2022-12-23T09:13:34.1673668Z 8 |         Err(e) if e.raw_os_error() == Some(libc::EISDIR) =>
2022-12-23T09:13:34.1673837Z   |                                                  ^^^^^^ not found in `libc`
2022-12-23T09:13:34.1673855Z
2022-12-23T09:13:34.1722461Z For more information about this error, try `rustc --explain E0425`.
2022-12-23T09:13:34.1768674Z error: could not compile `remove_dir_all` due to previous error
2022-12-23T09:13:34.1769130Z warning: build failed, waiting for other jobs to finish...
2022-12-23T09:13:36.3050960Z Error: Compiling your crate to WebAssembly failed
2022-12-23T09:13:36.3051416Z Caused by: failed to execute `cargo build`: exited with exit status: 101
2022-12-23T09:13:36.3052686Z   full command: "cargo" "build" "--lib" "--release" "--target" "wasm32-unknown-unknown" "--features" "console_error_panic_hook
```

I've moved some things so that `roc_test_utils` is a `dev_depencency` again and I've added a cfg around the use of remove_dir_all to ignore it for wasm.

### CODE REVIEW

File: crates/compiler/fmt/src/test_helpers.rs [hunk 1]
```diff
@@ -1,77 +1,82 @@
...
 
-            let mut buf = Buf::new_in(&arena);
+    /// Parse and re-format the given input, and pass the output to `check_formatting`
+    /// for verification.  The expectation is that `check_formatting` assert the result matches
+    /// expectations (or, overwrite the expectation based on a command-line flag)
+    /// Optionally, based on the value of `check_idempotency`, also verify that the formatting
+    /// is idempotent - that if we reformat the output, we get the same result.
+    pub fn expr_formats(input: &str, check_formatting: impl Fn(&str), check_idempotency: bool) {
+        let arena = Bump::new();
+        let input = input.trim();
```
Comment (Anton-4) [inline crates/compiler/fmt/src/test_helpers.rs:20]: This did not diff well, I just added:
```
#[cfg(test)]
pub mod test_helpers {
```

File: crates/compiler/fmt/src/test_helpers.rs [hunk 2]
```diff
@@ -1,77 +1,82 @@
...
 
-/// Parse and re-format the given input, and pass the output to `check_formatting`
-/// for verification.  The expectation is that `check_formatting` assert the result matches
-/// expectations (or, overwrite the expectation based on a command-line flag)
-/// Optionally, based on the value of `check_idempotency`, also verify that the formatting
-/// is idempotent - that if we reformat the output, we get the same result.
-pub fn expr_formats(input: &str, check_formatting: impl Fn(&str), check_idempotency: bool) {
-    let arena = Bump::new();
-    let input = input.trim();
+#[cfg(test)]
```
Comment (joshuawarner32) [inline crates/compiler/fmt/src/test_helpers.rs:3]: Making this `#[cfg(test)]` will break the fuzzing. FWIW, parse has its own `test_helpers.rs` that's not cfg'd like this.

What's the purpose of this change?

**Review by joshuawarner32 — CHANGES_REQUESTED** [review]

File: crates/compiler/fmt/src/test_helpers.rs (hunk 2 above)
Comment (Anton-4) [reply crates/compiler/fmt/src/test_helpers.rs:3]: It's so that `roc_test_utils` can be a dev-dependency, which it should always be. Trying to include it as a regular dependency caused problems for the wasm-repl. The `[target.'cfg(not(target_family = "wasm"))'.dependencies]` I added will also fix this. But I want to prevent unnecessary compilation and problems like this that are caused by unnecessary dependencies that are never used in the roc executable.

How can I run the fuzzing?

Comment (joshuawarner32) [reply crates/compiler/fmt/src/test_helpers.rs:3]: Ah, interesting.

Run fmt fuzzing like this:
```
cd crates/compiler/fmt
cargo +nightly fuzz run -j4 fuzz_expr -- -dict=fuzz/dict.txt
```

(also make sure you have a semi-recent nightly installed)

**Comment by Anton-4** [comment]
This was fixed in another PR
